arkane/common.py

```diff
@@ -231,21 +231,26 @@ def save_yaml(self, path):
             yaml.dump(data=self.as_dict(), stream=f)
         logging.debug('Dumping species {0} data as {1}'.format(self.label, filename))
 
-    def load_yaml(self, path, species, pdep=False):
+    def load_yaml(self, path, label=None, pdep=False):
         """
         Load the all statMech data from the .yml file in `path` into `species`
         `pdep` is a boolean specifying whether or not jobList includes a pressureDependentJob.
         """
-        logging.info('Loading statistical mechanics parameters for {0} from .yml file...'.format(species.label))
+        yml_file = os.path.basename(path)
+        if label:
+            logging.info('Loading statistical mechanics parameters for {0} from {1} file...'.format(label, yml_file))
+        else:
+            logging.info('Loading statistical mechanics parameters from {0} file...'.format(yml_file))
         with open(path, 'r') as f:
             data = yaml.safe_load(stream=f)
-        try:
-            if species.label != data['label']:
-                logging.debug('Found different labels for species: {0} in input file, and {1} in the .yml file. '
-                              'Using the label "{0}" for this species.'.format(species.label, data['label']))
-        except KeyError:
-            # Lacking label in the YAML file is strange, but accepted
-            logging.debug('Did not find label for species {0} in .yml file.'.format(species.label))
+        if label:
```

Comment:
You can move it to the `if label` condition above

Reply:
I thought about that, but then the "Loading ..." message will have to come after we load `data = yaml.safe_load(stream=f)`. Which would you prefer?